Source/carthage/Extensions.swift

```diff
@@ -47,6 +47,38 @@ internal func print<T>(object: T) {
 	}
 }
 
+internal final class FileHandle {
+	let fileDescriptor: Int32
+	let pointer: UnsafeMutablePointer<FILE>
+
+	init(fileDescriptor fd: Int32) {
+		fileDescriptor = fd
+		pointer = fdopen(fd, "r+")
+	}
+
+	func write(data: NSData) -> Result<Int> {
+		switch Darwin.write(fileDescriptor, data.bytes, UInt(data.length)) {
+		case (let value) where value != 0:
```

Comment:
This seems unnecessarily confusing. Why can't we assign to a temporary and use `if`? I think that would express our intention much more clearly.